backend/index.ts

```diff
@@ -35,3 +40,38 @@ const port = 8000;
 app.listen(8000, () => {
     console.log(`Server started: http://localhost:${port}/`);
 }); 
+
+app.get('/login/password',
+  function(req, res, next) {
+    res.render('login');
+  });
+
+app.post('/login/password',
+  passport.authenticate('local', { failureRedirect: '/login', failureMessage: true }),
+  function(req, res) {
+    console.log(req.body)
+    res.status(200).send("we logged in")
+  });
+
+const connection = mysql.createConnection({
+    host: process.env.DB_HOST,
+    user: process.env.DB_USER,
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_NAME
+  });
+  
+passport.use(new LocalStrategy(function verify(username:any, password:any, cb:any) {
+    connection.query('SELECT * FROM users WHERE username = ?', [ username ], function(err:any, user:any) {
+      if (err) { return cb(err); }
+      if (!user) { return cb(null, false, { message: 'Incorrect username or password.' }); }
+  
+      crypto.pbkdf2(password, user.salt, 310000, 32, 'sha256', function(err:any, hashedPassword:any) {
+        if (err) { return cb(err); }
+        if (!crypto.timingSafeEqual(user.hashed_password, hashedPassword)) {
+          return cb(null, false, { message: 'Incorrect username or password.' });
+        }
+        return cb(null, user);
+      });
```

Comment:
This is a little hard to read for me. We already have the verify function named so maybe we can break it up so that it's a little easier to read?